liquidctl/driver/nzxt_smart_device.py

```diff
@@ -182,66 +276,244 @@ def get_status(self, **kwargs):
+                break
+            msg = self.device.read(self._READ_LENGTH)
+            LOGGER.debug('received %s', ' '.join(format(i, '02x') for i in msg))
+            if not msg_1101_reply and msg[0] == 0x11 and msg[1] == 0x01:
+                fw = '{}.{}.{}'.format(msg[0x11], msg[0x12], msg[0x13])
+                status.append(('Firmware version', fw, ''))
+                num_valid_replies_recvd += 1
+                msg_1101_reply = True
+                continue
+            if not msg_2103_reply and msg[0] == 0x21 and msg[1] == 0x03:
+                num_light_channels = msg[14]  # the 15th byte (index 14) is # of light channels
+                accessories_per_channel = 6   # each lighting channel supports up to 6 accessories
+                light_accessory_index = 15    # offset in msg of info about first light accessory
+                for light_channel in range(num_light_channels):
+                    for accessory_num in range(accessories_per_channel):
+                        accessory_id = msg[light_accessory_index]
+                        light_accessory_index += 1
+                        if accessory_id != 0:
+                            status.append(('LED {} accessory {}'.format(light_channel + 1, accessory_num + 1),
+                                           self._ACCESSORY_NAMES.get(accessory_id, 'Unknown'), ''))
+                num_valid_replies_recvd += 1
+                msg_2103_reply = True
+                continue
+            if not msg_6702_reply and msg[0] == 0x67 and msg[1] == 0x02:
+                rpm_offset = 24
+                duty_offset = 40
+                noise_offset = 56
+                for i, _ in enumerate(self._speed_channels):
+                    if ((msg[rpm_offset] != 0x0) and (msg[rpm_offset + 1] != 0x0)):
+                        status.append(('Fan {} speed'.format(i + 1), msg[rpm_offset + 1] << 8 | msg[rpm_offset], 'rpm'))
+                        status.append(('Fan {} duty'.format(i + 1), msg[duty_offset + i], '%'))
+                    rpm_offset += 2
+                status.append(('Noise level', msg[noise_offset], 'dB'))
+                num_valid_replies_recvd += 1
+                msg_6702_reply = True
+                continue
+            if not msg_6704_reply and msg[0] == 0x67 and msg[1] == 0x04:
+            # PLACE HOLDER FOR FUTURE WORK
+            # Interpretation of this reply is pending
+                #status.append((' Unknown:', 'status', 'pending'))
+                num_valid_replies_recvd += 1
+                msg_6704_reply = True
+                continue
+        self.device.release()
+        return sorted(status)
+
+    def _write_colors(self, cid, mode, colors, sval):
+        mval, mod3, mod4, mincolors, maxcolors = self._COLOR_MODES[mode]
+        color_count = len(colors)
+        if maxcolors == 40:
+            channel_mod = [0x1E, 0x20][cid] # the purpose of this is unknown, but is based on cmd issued by CAM software
```

Comment:
It's weird that comparing this (`if maxcolors == 40`) and the final (`else`) cases, `channel_mod` is only different for channel led1.

Reply:
This is one of the redundancy checks I can do -- namely, determine whether `channel_mod` does in fact make any difference. If not, I'll remove it.

Reply:
Ok, channel_mod makes no difference. Removed.